### PR TITLE
Use memize when creating new object in component to avoid potential unnecessary rerendering

### PR DESCRIPTION
**Pull request by felixarntz**
## Summary

Addresses issue:

- #5987 

See https://github.com/google/site-kit-wp/issues/5987#issuecomment-1302300889 (follow up to #6088).

## Relevant technical choices

* Better be safe than sorry :)

## PR Author Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 5.2 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [ ] I have added a QA Brief on the issue linked above.
- [ ] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/components/ReportError.js [hunk 1]
```diff
@@ -68,12 +69,16 @@ export default function ReportError( { moduleSlug, error } ) {
...
+		};
+	} );
+
 	const showRetry = !! retryableErrors.length;
 
 	const errorTroubleshootingLinkURL = useSelect( ( select ) => {
-		const err = {
-			...( showRetry ? retryableErrors[ 0 ] : errors[ 0 ] ),
-		};
+		const err = determineError( retryableErrors, errors );
```
Comment (aaemnnosttv) [inline assets/js/components/ReportError.js:81]: If the concern here is to avoid unnecessary re-renders of the component, this shouldn't be necessary because `useSelect` only triggers a re-render when the output of the function is different. In this case, the output is a string, which wouldn't change based on the identity of the object.

It's also worth noting that `useSelect` uses a shallow equality comparison for this so even if a new object was returned from the `useSelect` every time, it wouldn't force a re-render on every invocation. See https://github.com/WordPress/gutenberg/blob/3e867dd7c6560e4b2cb9a59cc02856e055be6142/packages/data/src/components/use-select/index.js#L126-L138

The other concern here would be that a new `err` object passed to the `getErrorTroubleshootingLinkURL` selector below would invoke its resolver every time due to a "new" set of arguments it's called with. Internally it uses https://github.com/aduth/equivalent-key-map which uses deep equality but is loose in terms of the identity comparison, so this would also be okay!

The problem with new objects is more of a risk when used in other places which are more strict about the comparison like dependency arrays for hooks or even [arguments passed to a `memize`-d function](https://github.com/google/site-kit-wp/pull/2690) 🙂 

**TL;DR** – since the object is only used to produce a string from the `useSelect`, there should be no problem with the way it is now.

**Review by aaemnnosttv — CHANGES_REQUESTED** [review]
Thanks @felixarntz! This shouldn't be necessary (see explanation below) but if it was, there would be the one change we'd want to make.

**Comment by felixarntz** [comment]
Thanks @aaemnnosttv for clarifying, that makes sense. Closing this one then.
